sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/src/main/java/com/azure/messaging/eventhubs/checkpointstore/blob/BlobPartitionManager.java

```diff
@@ -0,0 +1,225 @@
+import java.util.concurrent.ConcurrentHashMap;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+/**
+ * Implementation of {@link PartitionManager} that uses
+ * <a href="https://docs.microsoft.com/en-us/azure/storage/common/storage-introduction#blob-storage">Storage Blobs</a>
+ * for persisting partition ownership and checkpoint information. {@link EventProcessor EventProcessors} can use this
+ * implementation to load balance and update checkpoints.
+ *
+ * @see EventProcessor
+ */
+public class BlobPartitionManager implements PartitionManager {
+
+    // All languages should use the same key for metadata. Discuss name for these.
+    private static final String SEQUENCE_NUMBER = "SequenceNumber";
+    private static final String OFFSET = "Offset";
+    private static final String OWNER_ID = "OwnerId";
+    private static final String ETAG = "eTag";
+
+    private static final String BLOB_PATH_SEPARATOR = "/";
+    private static final ByteBuffer UPLOAD_DATA = ByteBuffer.wrap("".getBytes(UTF_8));
+
+    private final ContainerAsyncClient containerAsyncClient;
+    private final ClientLogger logger = new ClientLogger(BlobPartitionManager.class);
+    private final Map<String, BlobAsyncClient> blobClients = new ConcurrentHashMap<>();
+
+    /**
+     * Creates an instance of BlobPartitionManager.
+     *
+     * @param containerAsyncClient The {@link ContainerAsyncClient} this instance will use to read and update blobs in
+     * the storage container.
+     */
+    public BlobPartitionManager(ContainerAsyncClient containerAsyncClient) {
+        this.containerAsyncClient = containerAsyncClient;
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param eventHubName The Event Hub name to get ownership information.
+     * @param consumerGroupName The consumer group name to get ownership information.
+     * @return A {@link Flux} of current partition ownership information.
+     */
+    @Override
+    public Flux<PartitionOwnership> listOwnership(String eventHubName, String consumerGroupName) {
+        String prefix = getBlobPrefix(eventHubName, consumerGroupName);
+        BlobListDetails details = new BlobListDetails().metadata(true);
+        ListBlobsOptions options = new ListBlobsOptions().prefix(prefix).details(details);
+        return containerAsyncClient.listBlobsFlat(options)
+            // Blob names should be of the pattern eventhub/consumergroup/<partitionId>
+            // While we can further check if the partition id is numeric, it may not necessarily be the case in future.
+            .filter(blobItem -> blobItem.name().split(BLOB_PATH_SEPARATOR).length == 3)
+            .map(this::convertToPartitionOwnership);
+    }
+
+    /**
+     * {@inheritDoc}
+     *
+     * @param requestedPartitionOwnerships Array of partition ownerships this instance is requesting to own.
+     * @return A {@link Flux} of successfully claimed ownership of partitions.
+     */
```

Comment:
I've not seen this pattern of inheritDoc + new docs. Can you please confirm you get the result you expect to see in the generated JavaDoc?

Reply:
Yeah, inheritDoc + new docs don't work well. So, just removed inheritDoc and added new docs for each method.